cairo/ethereum/cancun/trie.cairo

```diff
@@ -372,55 +464,353 @@ func bytes_to_nibble_list{bitwise_ptr: BitwiseBuiltin*}(bytes_: Bytes) -> Bytes
+    assert branches[4] = branches_4;
+    if (value_4.value.len != 0) {
+        assert value = value_4;
+        assert value_set = 1;
+    }
+    let (branches_5, value_5) = _get_branche_for_nibble_at_level(obj, 5, level.value);
+    assert branches[5] = branches_5;
+    if (value_5.value.len != 0) {
+        assert value = value_5;
+        assert value_set = 1;
+    }
+    let (branches_6, value_6) = _get_branche_for_nibble_at_level(obj, 6, level.value);
+    assert branches[6] = branches_6;
+    if (value_6.value.len != 0) {
+        assert value = value_6;
+        assert value_set = 1;
+    }
+    let (branches_7, value_7) = _get_branche_for_nibble_at_level(obj, 7, level.value);
+    assert branches[7] = branches_7;
+    if (value_7.value.len != 0) {
+        assert value = value_7;
+        assert value_set = 1;
+    }
+    let (branches_8, value_8) = _get_branche_for_nibble_at_level(obj, 8, level.value);
+    assert branches[8] = branches_8;
+    if (value_8.value.len != 0) {
+        assert value = value_8;
+        assert value_set = 1;
+    }
+    let (branches_9, value_9) = _get_branche_for_nibble_at_level(obj, 9, level.value);
+    assert branches[9] = branches_9;
+    if (value_9.value.len != 0) {
+        assert value = value_9;
+        assert value_set = 1;
+    }
+    let (branches_10, value_10) = _get_branche_for_nibble_at_level(obj, 10, level.value);
+    assert branches[10] = branches_10;
+    if (value_10.value.len != 0) {
+        assert value = value_10;
+        assert value_set = 1;
+    }
+    let (branches_11, value_11) = _get_branche_for_nibble_at_level(obj, 11, level.value);
+    assert branches[11] = branches_11;
+    if (value_11.value.len != 0) {
+        assert value = value_11;
+        assert value_set = 1;
+    }
+    let (branches_12, value_12) = _get_branche_for_nibble_at_level(obj, 12, level.value);
+    assert branches[12] = branches_12;
+    if (value_12.value.len != 0) {
+        assert value = value_12;
+        assert value_set = 1;
+    }
+    let (branches_13, value_13) = _get_branche_for_nibble_at_level(obj, 13, level.value);
+    assert branches[13] = branches_13;
+    if (value_13.value.len != 0) {
+        assert value = value_13;
+        assert value_set = 1;
+    }
+    let (branches_14, value_14) = _get_branche_for_nibble_at_level(obj, 14, level.value);
+    assert branches[14] = branches_14;
+    if (value_14.value.len != 0) {
+        assert value = value_14;
+        assert value_set = 1;
+    }
+    let (branches_15, value_15) = _get_branche_for_nibble_at_level(obj, 15, level.value);
+    assert branches[15] = branches_15;
+    if (value_15.value.len != 0) {
+        assert value = value_15;
+        assert value_set = 1;
+    }
+    %{ ids.value_set = memory.get(fp + 2) or 0 %}
+    if (value_set != 1) {
+        let (data: felt*) = alloc();
+        tempvar empty_bytes = Bytes(new BytesStruct(data, 0));
+        assert value = empty_bytes;
+    }
```

Comment:
(same as previous comment regarding value_set being wrongly returned by the prover and checks on existing memory rather than creating a new segment)